### ISSUE TITLE
Redundant npm packages included in client production bundle

### ISSUE THREAD

**Issue by arggh** [meteor/meteor#10484]
### Bad behavior

Using the latest beta version `1.8.1-beta.20`, I witnessed the following:

1. Take a look at bundle visualizer
2. Notice that one of my Meteor packages is including `lodash` in the final bundle
3. Double check that `lodash` is not actually depended upon by that package and not mentioned in `package.js`
4. Go to the package directory `.npm/package` and run `npm ls`
5. Find out that `lodash` is included as dependency of `babel-plugin-emotion`, which gets installed when I include `emotion` in the `package.js`. I don't think it should, since `babel-plugin-emotion` should be a dev dependency, if installed.
```
$ npm ls
my-package/.npm/package/
├─┬ emotion@10.0.7
│ ├─┬ babel-plugin-emotion@10.0.7
│ │ ├─┬ @babel/helper-module-imports@7.0.0
│ │ │ └─┬ @babel/types@7.3.4
│ │ │   ├── esutils@2.0.2
│ │ │   ├── lodash@4.17.11
│ │ │   └── to-fast-properties@2.0.0
```
6. `babel-plugin-emotion` depends on `lodash` and so that gets wrapped in the final app bundle

### How to fix

7. Remove `emotion` from package's npm dependencies
8. Add `emotion` as app's dependencies
9. Remove `.npm` from the package dir (otherwise `lodash` will still get bundled!)
10. Restart the app
11. `lodash` is gone from the production bundle

### Expected behavior

To not bundle a dev dependency of a npm package that's depended by the Meteor package

**Comment by arggh** [comment]
Updated the original issue post as I discovered more.

**Comment by benjamn** [comment]
Meteor only bundles what the package imports, so you shouldn't be seeing `emotion` or `babel-plugin-emotion` unless those packages are imported by the Meteor package (or imported by code that the Meteor package imports, and so forth). Can you dig into how that import is happening, given that `babel-plugin-emotion` should only be imported/used in development?

**Comment by arggh** [comment]
In the scenario I tried to describe above...

- `emotion` was included, as intended
- `babel-plugin-emotion` was not included, at least I didn't notice it would by inspecting the bundle analysis diagram
- `lodash` was included **twice**, once in the app, once in the package

I then checked what could possibly be responsible for installing lodash into the package directory, and the only suspect was `babel-plugin-emotion`. I'm not sure if it has anything to do with it, though. Maybe just a hasty conclusion on my part.

I made a reproduction that demonstrates the issue: https://github.com/arggh/meteor-pkg-issue

In the reproduction I created a Meteor app with `meteor create` and after that, created a new Meteor package, `testpkg`, which I configured to depend on `emotion` npm package. `testpkg` also uses a function from `lodash`, but the package itself doesn't depend (in `package.js`) on `lodash`. Instead, `lodash` is installed as a dependency of the app.

As a result, `lodash` is included twice in the production bundle, as demostrated by the bundle visualizer.
![Mar-13-2019 16-44-58](https://user-images.githubusercontent.com/17210302/54288785-b4b78f00-45b0-11e9-8f42-5b9fe37bd248.gif)

If you remove the `emotion` dependency from the Meteor package, and install `emotion` as a dependency of the app itself via `meteor npm install --save emotion@latest`, then `lodash` will get bundled only once.
